java/src/org/openqa/selenium/remote/AbstractDriverOptions.java

```diff
@@ -30,13 +35,47 @@
 import org.openqa.selenium.UnexpectedAlertBehaviour;
 import org.openqa.selenium.internal.Require;
 
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeMap;
 import java.util.TreeSet;
 
 public abstract class AbstractDriverOptions<DO extends AbstractDriverOptions> extends MutableCapabilities {
+  public DO setBrowserVersion(String browserVersion) {
+    setCapability(
+      BROWSER_VERSION,
+      Require.nonNull("Browser version", browserVersion));
+    return (DO) this;
+  }
+
+  public DO setPlatformName(String platformName) {
+    setCapability(
+      PLATFORM_NAME,
+      Require.nonNull("Platform Name", platformName));
+    return (DO) this;
+  }
+
+  public DO setTimeouts(Map<String, Duration> timeouts) {
+    HashMap<String, Long> convertedTimeouts = new HashMap<>();
+    List<String> validTimeouts = new ArrayList<>(Arrays.asList(IMPLICIT_TIMEOUT, PAGE_LOAD_TIMEOUT, SCRIPT_TIMEOUT));
+
+    Require.nonNull("Timeouts", timeouts).forEach((k, v) -> {
+      if (validTimeouts.contains(k)) {
+        convertedTimeouts.put(k, v.toMillis());
+      } else {
+        throw new IllegalArgumentException(k + " is not one of the valid timeout values: " + validTimeouts);
+      }
+    });
+
+    setCapability(TIMEOUTS, convertedTimeouts);
```

Comment:
I think this has the potential to get confusing --- somewhere in a framework the implicit timeout is set, and then user code sets the page load timeout: what do we expect to happen to the implicit timeout? I _think_ people will expect both to be set.

Reply:
splitting up the timeouts as suggested above fixes this issue nicely.